mochify/lib/server.js

```diff
@@ -25,11 +25,25 @@ async function startServer(base_path, options = {}) {
   };
 }
 
-function requestHandler(base_path) {
+function requestHandler(base_path, scripts = []) {
   return async (req, res) => {
     if (req.url === '/') {
       res.writeHead(200);
-      res.end('<!DOCTYPE html>\n<html><body></body></html>');
+      res.end(`
+<!DOCTYPE html>
+<html>
+${scripts
+  .map((script) => {
+    const { content = '', ...attrs } = script;
```

Comment:
Why did you come up with this "protocol"? Couldn't we add the generated script tag strings to the `scripts` array?

Reply:
> Why did you come up with this "protocol"? 

It's called overengineering and I am guilty :church: 

> Couldn't we add the generated script tag strings to the scripts array?

Then who is responsible for distinguishing a string like `./foo.test.js` and `alert('hello world')`? Even if we'd have `index.js` pass down strings the server could use as-is, someone would still need to know about the logic how to construct valid inline scripts and scripts that reference source files.

---

We could definitely make this a little less complicated though I guess. What do you think about passing `scripts` (for inline scripts) and `modules` (for source files)? This way we could still have the "template" figure out how to render these things (which is where I think this should happen).

Reply:
I changed it to work as described and I think it's a lot nicer now.